src/proto/FunctionRpc.proto

```diff
@@ -262,6 +267,51 @@ message RpcFunctionMetadata {
   bool is_proxy = 7;
 }
 
+// Host tells worker it is ready to receive metadata
+message WorkerFunctionMetadataRequest {
+  // base directory for function app
+  string directory = 1;
+}
+
+// Worker sends function metadata back to host
+message WorkerFunctionMetadataResponse {
```

Comment:
Intention of this message is to load functions indexed by the worker in the host?  
`FunctionsLoadRequest` that sends a list of `FunctionLoadRequest` ? As long as fields are not removed - reusing message types is possible and backward compatible.

Reply:
Good call, going to rename to `FunctionsLoadRequest` and `FunctionLoadResponses` because the initial message is requesting to load all functions in the root directory and the response contains multiple response objects.